packages/httpsnippet-client-api/src/index.ts

```diff
@@ -224,10 +226,9 @@ const client: Client<APIOptions> = {
             return;
           }
         } else if (headerLower === 'accept') {
-          // If the `Accept` header here is not the default or first `Accept` header for the
-          // operations' request body then we should add it otherwise we can let the SDK handle it
-          // itself.
-          if (headers[header] === operation.getContentType()) {
+          // If the `Accept` header here is JSON-like header then we can remove it from the code
+          // snippet because `api` natively supports and prioritizes JSON over any other mime type.
+          if (matchesMimeType.json(headers[header] as string)) {
```

Comment:
would we also want to delete the header in the case of `headers[header] === operation.getContentType()`? i feel like it makes sense intuitively to do so?

```suggestion
          if (matchesMimeType.json(headers[header] as string) || headers[header] === operation.getContentType()) {
```

Reply:
This was actually a bad assumption on my part equating `Accept` to `Content-Type`. We already handle `Content-Type` headers right above this.